backend/main.go

```diff
@@ -138,78 +200,144 @@ func roll(level int, rolls int) {
 			}
 
 			if c, ok := rollResults[champion]; ok {
-				rollResults[champion] = rollResult{champion, cost + 1, c.Appearances + 1}
+				rollResults[champion] = rollResult{champion, cost + 1, c.Appearances + 1, ""}
 			} else {
-				rollResults[champion] = rollResult{champion, cost + 1, 1}
+				rollResults[champion] = rollResult{champion, cost + 1, 1, ""}
 			}
 		}
 	}
+
+	// calculate the percentage of appearance for each champion
+	for i, s := range rollResults {
```

Comment:
Instead of mutating a global object, consider returning it.

```go
func roll(level, rolls int) map[string]rollResult {
    results := map[string]rollResult{}
    for i := 0 ;  < rolls; r++ {
       ....
       if c, ok := results[champion]; ok {
       
        ....
     }

     return results
}
```